### DISCUSSION TITLE
Get first x blog posts for usage within static page

### DISCUSSION THREAD

**Discussion by TheZoker** [facebook/docusaurus#4443] [Q&A]
Hi there,

I'm currently looking for a way to display part of a blog post on a static page (front page in this case).

Is there any way to get the first x blog posts?

Something like:
```js
import getBlogPosts from '@theme/Blog';

const blogPosts = getBlogPosts(5);
for(....)
```

Thanks!

**Comment by slorber** [comment]
Blog posts are not available on non-blog pages due to the modular nature of Docusaurus.

You can create a custom plugin that would read the /blog folder and inject the required data into a homepage component as props. It requires a bit of hacking to get this working, but definitively possible.

You can also generate a blogPosts.json as a pre-build step and require this file with the data you need, or provide it in your config customFields attribute
